x/archivedb/key.go

```diff
@@ -0,0 +1,76 @@
+package archivedb
+
+import (
+	"errors"
+	"math"
+)
+
+var errBadVersion = errors.New("invalid version")
+
+// Key
+//
+// The keys contains a few extra information alongside the given key. The key is
+// what is known outside of this internal scope as the key, but this struct has
+// more information compacted inside the Key to take advantage of natural
+// sorting. The inversed height is stored right after the Prefix, which is
+// MaxUint64 minus the desired Height. An extra byte is also packed inside the
+// key to indicate if the key is a deletion of any previous value (because
+// archivedb is an append only database).
+//
+// Any other property are to not serialized but they are useful when parsing a
+// Key struct from the database
+type Key struct {
```

Comment:
We should comment that the length of the prefix is encoded in the bytes and is required for the correctness of the iteration fetching (thought this was a bug at first)

Reply:
if the length of the key ends up encoded in the bytes does this make iteration difficult?

Reply:
The length of the key is not stored. The suffix is known (9 bytes: uint64 and bool), the prefix is unknown.